cider/tests/functionality/CiderFilterTest.cpp

```diff
@@ -276,9 +261,245 @@ TEST_F(CiderFilterSequenceTestBase, multiColEqualTest) {
+      "not_in_fp32_array.json");
+  // TODO: add in (str_1, str_2, str_3)
+}
+
+TEST_F(CiderFilterNullTestBase, integerNullFilterTest) {
+  assertQuery("SELECT col_1 FROM test WHERE col_1 < 77");
+  assertQuery("SELECT col_2 FROM test WHERE col_2 > 77");
+  assertQuery("SELECT col_3 FROM test WHERE col_3 <= 77");
+  assertQuery("SELECT col_4 FROM test WHERE col_4 >= 77");
+  assertQuery("SELECT col_1 FROM test WHERE col_1 IS NOT NULL AND col_1 < 77");
+  assertQuery("SELECT col_2 FROM test WHERE col_2 IS NOT NULL AND col_2 > 77");
+  assertQuery("SELECT col_3 FROM test WHERE col_3 IS NOT NULL AND col_3 <= 77");
+  assertQuery("SELECT col_4 FROM test WHERE col_4 IS NOT NULL AND col_4 >= 77");
+}
+
+// TODO: Comment this test out due to unsupported decimal, string, varchar
+// Update[0913, jikunshang]: decimal type is WIP. String and varchar filter have been
+// supported in CiderStringTest.
+
+TEST_F(CiderFilterNullTestBase, inTest) {
+  assertQuery("SELECT * FROM test WHERE col_1 in (24, 25, 26)", "in_int32_array.json");
+  assertQuery("SELECT * FROM test WHERE col_2 in (24, 25, 26)", "in_int64_array.json");
+  assertQuery("SELECT * FROM test WHERE col_3 in (24, 25, 26)", "in_fp32_array.json");
+  assertQuery("SELECT * FROM test WHERE col_4 in (24, 25, 26)", "in_fp64_array.json");
+  assertQuery("SELECT * FROM test WHERE col_1 IS NOT NULL AND col_1 in (24, 25, 26)");
+  assertQuery("SELECT * FROM test WHERE col_2 IS NOT NULL AND col_2 in (24, 25, 26)");
+  assertQuery("SELECT * FROM test WHERE col_3 IS NOT NULL AND col_3 in (24, 25, 26)");
+  assertQuery("SELECT * FROM test WHERE col_4 IS NOT NULL AND col_4 in (24, 25, 26)");
+  assertQuery("SELECT * FROM test WHERE col_3 not in (24, 25, 26)",
+              "not_in_fp32_array.json");
+  // TODO: add in (str_1, str_2, str_3)
+}
+
+TEST_F(CiderFilterRandomTestBase, DistinctFromTest) {
+  // IS DISTINCT FROM
+  assertQuery(
+      "SELECT * FROM test WHERE col_3 IS DISTINCT FROM col_7 OR col_4 IS DISTINCT FROM "
+      "col_8",
+      "is_distinct_from.json",
+      true);
+
+  // IS NOT DISTINCT FROM
+  assertQuery(
+      "SELECT * FROM test WHERE col_2 IS NOT DISTINCT FROM col_6 OR col_1 IS NOT "
+      "DISTINCT FROM col_5",
+      "is_not_distinct_from.json",
+      true);
+
+  // mixed case
+  assertQuery(
+      "SELECT * FROM test WHERE col_3 IS DISTINCT FROM col_7 OR col_1 IS NOT DISTINCT "
+      "FROM col_5",
+      "mixed_distinct_from.json",
+      true);
+
+  // mixed case with string
+  assertQuery(
+      "SELECT * FROM test WHERE col_9 IS DISTINCT FROM col_10 OR col_10 IS NOT DISTINCT "
+      "FROM col_9",
+      "mixed_distinct_from_string.json",
+      true);
+}
+
+// Extends CiderTestBase and create a (99 rows, 4 types columns) table for filter test.
+class CiderFilterSequenceTestArrow : public CiderTestBase {
+ public:
+  CiderFilterSequenceTestArrow() {
+    table_name_ = "test";
+    create_ddl_ =
+        R"(CREATE TABLE test(col_1 INTEGER, col_2 BIGINT,
```

Comment:
Please add `NOT NULL` in create table command if you didn't generate null data. generated IR will be different.

Reply:
done